libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/speech/MapboxSpeechPlayer.java

```diff
@@ -0,0 +1,233 @@
+package com.mapbox.services.android.navigation.ui.v5.voice.speech;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.media.MediaPlayer;
+import android.os.AsyncTask;
+import android.text.TextUtils;
+
+import com.amazonaws.auth.CognitoCachingCredentialsProvider;
+import com.mapbox.mapboxsdk.Mapbox;
+import com.mapbox.services.android.navigation.ui.v5.voice.InstructionListener;
+import com.mapbox.services.android.navigation.ui.v5.voice.InstructionPlayer;
+import com.mapbox.services.android.navigation.v5.navigation.VoiceInstructionLoader;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import okhttp3.Cache;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import timber.log.Timber;
+
+/**
+ * <p>
+ * Will retrieve synthesized speech mp3s from Mapbox's API Voice.
+ * </p>
+ */
+public class MapboxSpeechPlayer implements InstructionPlayer {
+  private static final long TEN_MB_CACHE_SIZE = 10 * 1098 * 1098;
```

Comment:
Minor minor detail.
Could you rename it to `TEN_MEGABYTE_CACHE_SIZE`? `MB` could be confused with Megabit.